Lagt til utenFeilContext for optional propagering av feil til barneko…

KenAJoh opened the pull request:
…mponenter

Eksempel for bruk av dette lagt til i Guideline-app

KenAJoh committed:
Lagt til utenFeilContext for optional propagering av feil til barnekomponenter

Eksempel for bruk av dette lagt til i Guideline-app

Lillebo commented on packages/node_modules/nav-frontend-skjema/src/skjema-gruppe.tsx:
```diff
@@ -6,8 +6,8 @@ import SkjemaelementFeilmelding from './skjemaelement-feilmelding';
-    'skjemagruppe--feil': harFeil
+const cls = (className, harFeil, utenContext) => classNames('skjemagruppe', className, {
+    'skjemagruppe--feil': harFeil && !utenContext
 });
```
Jeg tror du har rett i at disse er overflødige og kan fjernes 👍

Lillebo commented on packages/node_modules/nav-frontend-skjema/src/skjema-gruppe.tsx:
```diff
@@ -76,9 +88,7 @@ class SkjemaGruppe extends React.Component<SkjemaGruppeProps> {
-                <SkjemaGruppeFeilContext.Provider value={{ feil, feilmeldingId }}>
-                    {children}
-                </SkjemaGruppeFeilContext.Provider>
+                {propegateErrorToChildren}
```
Lurer på om det kanskje hadde vært mer lesbart om vi heller skrev noe ala 

```tsx
{ (utenFeilContext) ? children : childrenWithContext }
```

her? 🤔 Da kan du endre `propegateErrorToChildren` på linje 73 til å sette `childrenWithContext` heller.

Lillebo commented on packages/node_modules/nav-frontend-skjema/src/skjema-gruppe.tsx:
```diff
@@ -61,9 +65,17 @@ class SkjemaGruppe extends React.Component<SkjemaGruppeProps> {
             description,
             tag = 'fieldset',
             feilmeldingId,
```
Virker som vi har glemt å gi denne en default verdi. Bør kanskje endre denne til `feilmeldingId = guid()` 🤔Da bør vel også linje 103 oppdateres til `'aria-errormessage': (!!feil) ? feilmeldingId : undefined`

Lillebo commented on packages/node_modules/nav-frontend-skjema/md/skjemagruppe/Skjemagruppe.overview.mdx:
```diff
@@ -135,6 +135,26 @@ Skjemagrupper brukes vanligvis for å gruppere flere radioknapper eller checkbox
+Bruk `utenFeilContext`-propen for å bestemme om `feil`-propen til skjermgruppen skal propageres til barnekomponenter.
+
+<Example>
+    <SkjemaGruppe feil="Her er det noe feil" errorId={guid()} utenFeilContext>
```
Virker som det har sneket seg med en liten feil i det forrige eksempelet med "Felles feilmelding" og bruk av `errorId` der. Litt ref. kommentaren om default verdi på `feilmeldingId` så kan du fjerne `errorId` fra begge disse 🙂

Lillebo reviewed (changes requested):
Knallbra PR 👍Se kommentarer:

nutgaard commented on packages/node_modules/nav-frontend-skjema/src/skjema-gruppe.tsx:
```diff
@@ -126,13 +136,22 @@ class SkjemaGruppe extends React.Component<SkjemaGruppeProps> {
+    /**
+     * Option om man sernder feilcontext videre til barnekomponenter
+     */
+    utenFeilContext: PT.bool
```
Kan vi kalle denne noe alà `propagateErrorToChildren` eller noe også? "feilContext" hinter til hvor det konkret er implementert i komponenten, men er ikke nødvendigvis lett å forstå for noen som skal bruke den,

Lillebo replied:
Godt poeng 👍

KenAJoh replied:
Foreslår at vi holder props til Norsk intil videre?
Foreslår `propagereErrorTilBarn` som helt norsk
eller `propagereErrorTilChildren` som gir mer mening for react.

Lillebo replied:
Synes kanskje at default oppførsel her fortsatt bør være at komponenten propagerer dette til children? Da bør vi kanskje kalle denne propen `utenFeilPropagering` eller noe i den duren?

KenAJoh replied:
Enig. `utenFeilPropagering` bør være deskriptiv nok, siden bruk av prop blir forklart med beskrivelse og eksempel på design.nav.no.

KenAJoh committed:
Utbedring av skjema-gruppe tsx og mdx

Endret prop `utenFeilContext` -> `propagereFeilTilChildren`. Grunn: Bedre forståelse av funskjon for brukere.
Fjernet `errorId={guid()}` . Grunn: Prop ikke brukt, evt erstattet av `feilmeldingId`
Fjernet `skjemagruppe--feil` CSS. Grunn: Redundant siden child komponenter vil sette feil-styling selv.
Endret hvordan rendring av `Children` blir gjort. Grunn: Bedre lesbarhet
Endret setting av `aria-errormessage`. Grunn: Defaulter ikke til `undefined`etter endringer.

KenAJoh committed:
Endret navn på `propagereFeilTilChildren` prop -> `utenFeilPropagering`

Bedre forståelse av at default er at feil blir propagert til children

Lillebo commented on packages/node_modules/nav-frontend-skjema/md/skjemagruppe/Skjemagruppe.overview.mdx:
````diff
@@ -135,6 +135,26 @@ Skjemagrupper brukes vanligvis for å gruppere flere radioknapper eller checkbox
 </SkjemaGruppe>
 ```
 
+## Felles feilmelding uten propagering av feil til barnekomponenter
````
Foreslår å endre denne til en `h3` som er underordnet den forrige overskriften "Felles feilmelding". Da kan vi også forenkle ordlyden en smule - noe ala:

```
### Uten propagering av feil til children
```

KenAJoh replied:
![h3_endring](https://user-images.githubusercontent.com/26967723/78554556-a798be00-780b-11ea-8fa0-e26bf81d1605.JPG)

Noe slikt?

Lillebo replied:
Jau 👍 Men synes vi gjerne kan skrive "children" i stedet for "barnekomponenter". Ser egentlig ikke noe grunn til å oversette tekniske begreper som dette til norsk. Vi bruker f.eks. allerede engelsk på det tekniske begrepet "Legend" litt lengre opp på siden 🙂

KenAJoh replied:
Enig. Fjerner all bruk av "barnekomponenter" over "children". 

![h3_endring_2](https://user-images.githubusercontent.com/26967723/78556609-905bcf80-780f-11ea-963a-3a564eab929e.jpg)

KenAJoh committed:
Formatering av mdx, "barnekomponenter" -> "children"

Lillebo reviewed (approved):
🎉
